plugins/woocommerce-blocks/assets/js/atomic/blocks/product-elements/summary/edit.tsx

```diff
@@ -1,33 +1,223 @@
-import {
-	BLOCK_TITLE as label,
-	BLOCK_ICON as icon,
-	BLOCK_DESCRIPTION as description,
-} from './constants';
+import { useIsDescendentOfSingleProductBlock } from '../shared/use-is-descendent-of-single-product-block';
+import { useIsDescendentOfSingleProductTemplate } from '../shared/use-is-descendent-of-single-product-template';
+import type { EditProps, ControlProps } from './types';
 import './editor.scss';
-import type { BlockAttributes } from './types';
 
-interface Props {
-	attributes: BlockAttributes;
-}
+const ShowDescriptionIfEmptyControl = ( {
+	showDescriptionIfEmpty,
+	setAttributes,
+}: ControlProps< 'showDescriptionIfEmpty' > ) => {
+	const label = __( 'Show description if empty', 'woocommerce' );
+	const help = __(
+		"Display the product description if it doesn't have a summary",
+		'woocommerce'
+	);
+
+	return (
+		<ToolsPanelItem
+			label={ label }
+			hasValue={ () => showDescriptionIfEmpty === true }
+			onDeselect={ () =>
+				setAttributes( { showDescriptionIfEmpty: false } )
+			}
+			isShownByDefault
+		>
+			<ToggleControl
+				label={ label }
+				help={ help }
+				checked={ showDescriptionIfEmpty }
+				onChange={ ( value ) => {
+					setAttributes( {
+						showDescriptionIfEmpty: value,
+					} );
+				} }
+			/>
+		</ToolsPanelItem>
+	);
+};
+
+const MaxWordCountControl = ( {
+	summaryLength,
+	setAttributes,
+}: ControlProps< 'summaryLength' > ) => {
+	const label = __( 'Max word count', 'woocommerce' );
+	const help = __( 'Set to 0 to remove the limit completely', 'woocommerce' );
+
+	return (
+		<ToolsPanelItem
+			label={ label }
+			hasValue={ () => summaryLength !== 0 }
+			onDeselect={ () => setAttributes( { summaryLength: 0 } ) }
+			isShownByDefault
+		>
+			<RangeControl
+				label={ label }
+				help={ help }
+				value={ summaryLength }
+				onChange={ ( value ) => {
+					setAttributes( {
+						summaryLength: value || 0,
+					} );
+				} }
+				min={ 0 }
+				max={ 200 }
+				step={ 10 }
+			/>
+		</ToolsPanelItem>
+	);
+};
+
+const LinkToDescriptionControl = ( {
+	showLink,
+	setAttributes,
+}: ControlProps< 'showLink' > ) => {
+	const label = __( 'Link to description', 'woocommerce' );
+	const help = __(
+		"Display a button to let shoppers jump to the product's description",
+		'woocommerce'
+	);
 
-const Edit = ( { attributes }: Props ): JSX.Element => {
+	return (
+		<ToolsPanelItem
+			label={ label }
+			hasValue={ () => showLink === false }
+			onDeselect={ () => setAttributes( { showLink: false } ) }
+			isShownByDefault
+		>
+			<ToggleControl
+				label={ label }
+				help={ help }
+				checked={ showLink }
+				onChange={ ( value ) => {
+					setAttributes( {
+						showLink: value,
+					} );
+				} }
+			/>
+		</ToolsPanelItem>
+	);
+};
+
+const LinkToDescription = ( {
+	linkText,
+	setAttributes,
+}: ControlProps< 'linkText' > ) => {
+	return (
+		<p>
+			<RichText
+				identifier="linkToDescrption"
+				className="wp-block-post-excerpt__more-link"
```

Comment:
Should we update the className?

Reply:
Definitely! Thanks for catching this! Updated in https://github.com/woocommerce/woocommerce/pull/48984/commits/aa295a87b1c130f2073e4865fe997f43efbd9282